agent/consul/server.go

```diff
@@ -1676,6 +1689,19 @@ func (s *Server) ReloadConfig(config ReloadableConfig) error {
 	}
 
 	s.rpcLimiter.Store(rate.NewLimiter(config.RPCRateLimit, config.RPCMaxBurst))
+
+	s.incomingRPCLimiter.UpdateConfig(rpcRate.HandlerConfig{
+		GlobalMode: config.RequestLimitsMode,
+		GlobalReadConfig: multilimiter.LimiterConfig{
+			Rate:  config.RequestLimitsReadRate,
+			Burst: int(config.RequestLimitsReadRate) * requestLimitsBurstMultiplier,
+		},
+		GlobalWriteConfig: multilimiter.LimiterConfig{
+			Rate:  config.RequestLimitsWriteRate,
+			Burst: int(config.RequestLimitsWriteRate) * requestLimitsBurstMultiplier,
+		},
+	})
```

Comment:
This is fine for now but I'm wondering if we need to check if the config changes before updating it. This could be done in the multi-limiter to make an update config a no-op if the config don't change.

Reply:
Good point. Would this be a comparison with the handler's config?